src/backend/EdFi.DataManagementService.Backend.Postgresql.Test.Integration/UpdateTests.cs

```diff
@@ -657,6 +659,94 @@ public void It_should_be_a_successful_update_for_2nd_transaction_due_to_retry()
         }
     }
 
+    [TestFixture]
+    public class Given_an_update_of_a_document_cascade_to_parents : UpdateTests
+    {
+        private static readonly Guid _sessionDocumentUuid = Guid.NewGuid();
+        private static readonly Guid _sessionReferentialIdUuid = Guid.NewGuid();
+
+        private static readonly Guid _courseOfferingDocumentUuid = Guid.NewGuid();
+        private static readonly Guid _courseOfferingReferentialIdUuid = Guid.NewGuid();
+
+        [SetUp]
+        public async Task Setup()
+        {
+            //The document that will be referenced "Session"
+            IUpsertRequest sessionUpsertRequest = CreateUpsertRequest(
+                "Session",
+                _sessionDocumentUuid,
+                _sessionReferentialIdUuid,
+                """
+                {
+                    "sessionName": "Third Quarter"
+                }
+                """,
+                allowIdentityUpdates: true
+            );
+            var sessionUpsertResult = await CreateUpsert()
+                .Upsert(sessionUpsertRequest, Connection!, Transaction!, traceId);
+            sessionUpsertResult.Should().BeOfType<UpsertResult.InsertSuccess>();
+
+            IUpsertRequest courseOfferingUpsertRequest = CreateUpsertRequest(
+                "CourseOffering",
+                _courseOfferingDocumentUuid,
+                _courseOfferingReferentialIdUuid,
+                """
+                {
+                    "localCourseCode": "ABC",
+                    "sessionReference": {
+                        "sessionName": "Third Quarter"
+                    }
+                }
+                """,
+                CreateDocumentReferences(
+                    [new("CourseOffering", sessionUpsertRequest.DocumentInfo.ReferentialId.Value)]
+                )
+            );
+
+            var courseOfferingUpsertResult = await CreateUpsert()
+                .Upsert(courseOfferingUpsertRequest, Connection!, Transaction!, traceId);
+            courseOfferingUpsertResult.Should().BeOfType<UpsertResult.InsertSuccess>();
+        }
+
+        [Test]
+        public async Task Update_session()
+        {
+            var documentIdentityElement = new DocumentIdentityElement(
+                new JsonPath("$.sessionName"),
+                "Fourth Quarter"
+            );
+            IUpdateRequest sessionUpdateRequest = CreateUpdateRequest(
+                "Session",
+                _sessionDocumentUuid,
+                _sessionReferentialIdUuid,
+                """
+                {
+                    "sessionName": "Fourth Quarter"
+                }
+                """,
+                documentIdentityElements: [documentIdentityElement]
+            );
+
+            var sessionUpdateResult = await CreateUpdate()
+                .UpdateById(sessionUpdateRequest, Connection!, Transaction!, traceId);
+
+            sessionUpdateResult.Should().BeOfType<UpdateResult.UpdateSuccess>();
```

Comment:
Suggest moving this into `Setup`, and rename `Update_session` to something like `It_should_update_the_body_of_the_referencing_document`